src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -478,14 +474,30 @@ def fuzz(self,
     """LibFuzzerCommon.fuzz override."""
     self._test_ssh()
 
-    #TODO(flowerhack): Pass libfuzzer args (additional_args) here
+    logs.log('SSH works; sync corpus directories')
+    # Push corpus directories to the device.
+    data_dst = self.fuzzer.data_path('corpus')
+    for corpus in corpus_directories:
+      for corpfile in os.listdir(corpus):
+        self.fuzzer.device.store(os.path.join(corpus, corpfile), data_dst)
+
+    logs.log('Corpus synced; run fuzzer')
+    # Run the fuzzer.
     return_code = self.fuzzer.start(additional_args)
     self.fuzzer.monitor(return_code)
     self.process_logs_and_crash(artifact_prefix)
 
     with open(self.fuzzer.logfile) as logfile:
       symbolized_output = logfile.read()
 
+    logs.log('Fuzzer ran; pull down corpus')
+    # Pull corpus directories from the device.
+    data_src = self.fuzzer.data_path('corpus/*')
+    # By convention, the *first* corpus directory in the list is always where
```

Comment:
to clarify: you should be pulling from the first corpus on the VM onto the first corpus on the host, rather than everything in the first corpus.

Reply:
Ok, how do we define "first" corpus on the VM?  (Is it first lexicographically?)

And to make sure I understand: You're saying I should pull the *first corpus* on the VM into the *first corpus* on the host?

(Once I'm sure I understand I'll update accordingly!)

Reply:
first one provided in the list of arguments. when libFuzzer fuzzes,

e.g. `./fuzzer corpus1 corpus2`

All new units are written to the first provided corpus (i.e. corpus1). 

For the Fuchsia Qemu runner, we want to make this appear as if everything was happening on the host, hence why we should copy back items from the first (in the list of arguments) back to the corresponding directory on the host.

Reply:
alright, done!  should be more clear what it's doing now (added comments too)